src/Generators/SubProfileGenerator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\lightning\Generators;
+
+use DrupalCodeGenerator\Command\BaseGenerator;
+use DrupalCodeGenerator\Utils;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+class SubProfileGenerator extends BaseGenerator
+{
+
+  protected $name = 'lightning-subprofile';
+  protected $description = 'Generates a Lightning Subprofile.';
+  protected $alias = 'lsp';
+  protected $templatePath = __DIR__;
+  protected $destination = 'profiles';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function interact(InputInterface $input, OutputInterface $output)
+  {
+    $questions['name'] = new Question('Profile Name');
+    $questions['name']->setValidator([Utils::class, 'validateRequired']);
+    $questions['machine_name'] = new Question('Profile Machine Name');
+    $questions['machine_name']->setValidator([Utils::class, 'validateMachineName']);
+    $questions['description'] = new Question('Enter the description (optional)', NULL);
+    $questions['install'] = new Question('Additional modules to include (optional), separated by commas (e.g. context, rules, file_entity)', NULL);
+    $questions['install']->setNormalizer(function ($answer) {
+      return $answer ? array_map('trim', explode(',', $answer)) : [];
+    });
+    $questions['exclude'] = new Question('Lightning components to exclude (optional), separated by commas (e.g. lightning_workflow, lightning_media)', Null);
+    $questions['exclude']->setNormalizer(function ($answer) {
+      return $answer ? array_map('trim', explode(',', $answer)) : [];
+    });
+
+    $vars = &$this->collectVars($input, $output, $questions);
+
+    $this->addFile()
```

Comment:
I think we actually won't need to use a template for the info file at all. If we build an info array, I think we can do `$this->addFile()->path('...')->content(\Drupal\Component\Serialization\Yaml::encode($info_array));`